daemons/deploy_aws_elasticsearch.sh

```diff
@@ -0,0 +1,145 @@
+		sleep 5
+	done
+	echo Timeout waiting for Elasticsearch domain to be removed
+	return 1
+}
+
+function wait_for_elasticsearch_endpoint() {
+	timeout_value=$1
+	start_time=`date +'%s'`
+	timeout_expiration=$(($start_time + $timeout_value))
+
+	while [[ `date +'%s'` -lt $timeout_expiration ]]; do
+	    if [[ `get_elasticsearch_endpoint $elasticsearch_domain_name` ]]; then
+	        echo Elasticsearch endpoint is available.
+	        return 0
+	    fi
+        echo Waiting for Elasticsearch endpoint to be become available ...
+		sleep 5
+	done
+	echo Timeout waiting for Elasticsearch endpoint
+	return 1
+}
+
+function get_elasticsearch_arn() {
+    check_elasticsearch_domain_exists &&
+    aws es describe-elasticsearch-domain --domain-name $elasticsearch_domain_name 2> /dev/null \
+        | jq -r '.DomainStatus.ARN'
+}
+
+function get_elasticsearch_endpoint() {
+    check_elasticsearch_domain_exists &&
+    aws es describe-elasticsearch-domain --domain-name $elasticsearch_domain_name 2> /dev/null \
+        | jq -r '.DomainStatus.Endpoint'
+}
+
+function display_elasticsearch_domain_info() {
+    elasticsearch_arn=$(get_elasticsearch_arn $elasticsearch_domain_name)
+    echo "elasticsearch ARN: " $elasticsearch_arn
+
+    elasticsearch_endpoint=$(get_elasticsearch_endpoint $elasticsearch_domain_name)
+    echo "elasticsearch endpoint: " $elasticsearch_endpoint
+}
+
+function setup_elasticsearch_domain() {
+    if [[ $delete_existing_es_instance == True ]]; then
+        check_elasticsearch_domain_exists
+        if [[ $? -eq 0 ]] ; then
+            # An elasticsearch domain with this name already exists
+
+            echo Elasticsearch domain already exists for $elasticsearch_domain_name
+            echo Deleting existing Elasticsearch domain
+            delete_elasticsearch_domain $elasticsearch_domain_name
+
+            wait_for_elasticsearch_domain_removed 300
+       fi
+    fi
+
+    check_elasticsearch_domain_exists
+    if [[ $? -ne 0 ]]; then
+        echo Creating new Elasticsearch domain with name: $elasticsearch_domain_name
+        create_elasticsearch_domain $elasticsearch_domain_name
+
+        # Should we only update the policy when the the elasticsearch domain is newly created
+        # or always update the policy in case the policy has changed?
+        echo Setting/updating Elasticsearch policy
+        update_elasticsearch_domain_policy $elasticsearch_domain_name
+    else
+        echo Using existing Elasticsearch domain with name: $elasticsearch_domain_name
+    fi
+}
+
+#
+# Main
+#
+
+if [[ $# != 2 ]]; then
+    echo "Usage: $(basename $0) daemon-name stage"
+    exit 1
+fi
+daemon_name=$1 stage=$2
+elasticsearch_domain_name="${daemon_name}-${stage}"
+
+delete_existing_es_instance=False
+
+setup_elasticsearch_domain $elasticsearch_domain_name
+
+wait_for_elasticsearch_endpoint 300
+
+# TODO Delete existing indexes here via Python script
+
+display_elasticsearch_domain_info
+
+echo Completed Elasticsearch setup
```

Comment:
General Bash style issue, if you're going to be writing this much shellcode, I would prefer that you not pass variables out-of-band and instead pass them as arguments e.g. rewrite
```
foo() {
    echo $x
}
x=1
foo
```
as
```
foo() {
    local x=$1
    echo $x
}
ix=5
foo $ix
```